skytemple_randomizer/randomizer/util/util.py

```diff
@@ -488,6 +488,14 @@
 ]
 
 
+_str_file_cache: dict[Pmd2Language, Str] = {}
```

Comment:
This is a simple dict-based cache for `Str` objects (similar to `_ssb_file_cache` below). I've thought to add this as it ensures that when e.g. `get_all_string_files` is called multiple times, it doesn't have to deserialise it from the ROM files and can also carry on any modifications applied in previous stages.